src/rendergraph/opengl/backend/attributeset.cpp

```diff
@@ -0,0 +1,11 @@
+#include "rendergraph/attributeset.h"
+
+using namespace rendergraph::backend;
+
+AttributeSet::AttributeSet(std::initializer_list<Attribute> list,
+        const std::vector<QString>& names) {
+    int i = 0;
+    for (auto item : list) {
+        m_attributes.push_back(Attribute{item.m_tupleSize, item.m_primitiveType, names[i++]});
+    }
```

Comment:
I agree it needs bounds checking, but it should probably throw, because things will go wrong if the attributes are not properly named. (It would have saved me some time when I didn't understand when some shader wasn't rendering, I should have added the throw there and then!)

But ideally this should be a compile time check... Or maybe the initialiser list should contain std::pair's of attribute and name.

Reply:
yeah not sure. We could also try coming up with a codepath that does this all at compile time. something like 
```cpp
template<typename... Ts, int N>
makeAttributeSet(QString (&arr)[N]) {
    static_assert(sizeof...(Ts) == N);
    ...
}
```
could work. I think we need to distance ourselves from QString then though because doing compile-time stuff with that is virtually impossible from my experience.

Reply:
Making as many parts as possible `constexpr` would lay good groundwork for that because we know that all of the stuff that technically can be done `constexpr` we can also force to compile time with appropriate `static_assert`s.